### ISSUE TITLE
Setting up a Route53 association with a zone in another account fails

### ISSUE THREAD

**Issue by hashibot** [terraform-providers/terraform-provider-aws#585]
_This issue was originally opened by @FransUrbo as hashicorp/terraform#12465. It was migrated here as part of the [provider split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/). The original body of the issue is below._

<hr>

Specifying a `provider` in the `aws_route53_zone_association` resource doesn't work [as expected].

### Terraform Version
0.8.8

### Affected Resource(s)
Please list the resources as a list, for example:
- aws_route53_zone_association

### Terraform Configuration Files
```hcl
provider "aws" {
  region  = "eu-west-1"
  profile = "core"
  alias   = "core"
}

provider "aws" {
  region  = "eu-west-1"
  profile = "test"
}

resource "aws_vpc" "test" {
  # ...
}

resource "aws_instance" "test" {
  # ...
}

resource "aws_route53_zone_association" "test" {
  vpc_id  = "${aws_vpc.test.id}"
  zone_id = "${data.terraform_remote_state.base.zone_id_private}"
  provider      = "aws.core"
}
```

### Steps to Reproduce
1. Create two accounts
2. Setup a `aws_vpc` in each account
3. Setup an `aws_instance` in each VPC
4. [possibly setup VPC peering between the VPCs in the two accounts]
5. Setup a Route53 zone in the first account
6. `aws_route53_zone_association` the VPC in the first account with this zone.
7. Try to `aws_route53_zone_association` the VPC in the *second* account with the zone.

**Comment by kaijparo** [comment]
any update on this one?

**Comment by plaplanche** [comment]
I'm also interested by this

**Comment by julianalves** [comment]
The PR is open, I am also interested. Thanks for the hard work!

**Comment by luisamador** [comment]
I've got exactly the same issue. Would be good to fix this. Thanks!

**Comment by aggallim** [comment]
this would help a lot of people out. Many AWS customers are adoption a multi-account structure, this would help create a global namespace versus a very nasty AWS CLI automation..

**Comment by vchan2002** [comment]
+1 on this..... Having a Route 53 zone association isn't 100% sufficient because we have uses cases where we have ELBs in govcloud that needs to create a R53 record in our primary AWS account.  Govcloud does not have R53 so it's not possible to do zone associations.

**Comment by ewbankkit** [comment]
Note [this comment](https://github.com/terraform-providers/terraform-provider-aws/issues/384#issuecomment-357049374) about a problem that still remains for full automation even if the changes in https://github.com/terraform-providers/terraform-provider-aws/pull/2005 are merged.

**Comment by ChadScott** [comment]
@antonosmond even after doing that I get an access denied error. Is the resource maybe querying the API for something that's not visible to the remote account? I haven't dug into the code to look. The error I get is:
`AccessDenied: User: arn:aws:iam::<redacted> is not authorized to access this resource` where the ARN is the service account Terraform runs as. The authorization is present (in fact, I can do the association from the CLI).

**Comment by tdmalone** [comment]
Related: #384 and #8593

**Comment by bflad** [comment]
Support for cross-account Route 53 VPC Associations via a new `aws_route53_vpc_association_authorization` resource and updated `aws_route53_zone_association` resource handling has been merged and will release with version 3.1.0 of the Terraform AWS Provider, later today. Thanks to @goakley and @RyanJarv for implementation. 👍
